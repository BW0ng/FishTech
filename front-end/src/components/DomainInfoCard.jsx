
import { Accordion } from "react-bootstrap";

function DomainInfoCard(props) {
    return (
        <Accordion.Item eventKey="0">
            <Accordion.Header>Domain Information</Accordion.Header>
            <Accordion.Body>
                <strong>Name: </strong>
                {props.data.domainName}

                <br/>
                <br/>

                
                <strong>Domain Status: </strong>
                <br/>
                { props.data.status && props.data.status.split(" ").map(status => {
                    return <span>&emsp;{status}<br/></span>;
                })}

                <br/>
                <br/>

                <strong>Nameservers: </strong>
                <br/>
                {props.data.nameServers && props.data.nameServers.hostNames.map(servers => {
                     return <span>&emsp;{servers}<br/></span>;
                })}

                <br/>
                <br/>

                <h3>Dates</h3>
                <strong>Registry Expiration: </strong>
                {props.data.expiresDate}

                <br/>
                <br/>
                <strong>Registry Expiration: </strong>
                {props.data.createdDate}
                
                <br/>

                
            </Accordion.Body>
        </Accordion.Item>
    );
}

export default DomainInfoCard;