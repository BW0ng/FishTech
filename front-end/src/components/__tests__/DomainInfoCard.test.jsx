import React from 'react';
import renderer from 'react-test-renderer';
import DomainInfoCard from '../DomainInfoCard.jsx';

describe('DomainInfoCard', () => {
    it('renders correctly with full data output', () => {
        let input = { 
            domainName: 'DomainName',
            status: 'status1 status2 status3 status4',
            nameServers : { hostNames: ['server1', 'server2', 'server3', 'server4' ]},
            expiresDate : '2021-09-20T20:37:09.391Z',
            createdDate : '2021-09-20T20:37:09.391Z',

         };
        const component = renderer.create(<DomainInfoCard data={input} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        // On the first run of this test, Jest will generate a snapshot file automatically.
      });

      describe ('renders correctly', () => {
            it('with missing statuses', () => {
                let input = { 
                    domainName: 'DomainName',
                    nameServers : { hostNames: ['server1', 'server2', 'server3', 'server4' ]},
                    expiresDate : '2021-09-20T20:37:09.391Z',
                    createdDate : '2021-09-20T20:37:09.391Z',
        
                };
                const component = renderer.create(<DomainInfoCard data={input} />);
                let tree = component.toJSON();
                expect(tree).toMatchSnapshot();
                // On the first run of this test, Jest will generate a snapshot file automatically.
            });
            it('with missing one status', () => {
                let input = { 
                    domainName: 'DomainName',
                    status: 'status1',
                    nameServers : { hostNames: ['server1', 'server2', 'server3', 'server4' ]},
                    expiresDate : '2021-09-20T20:37:09.391Z',
                    createdDate : '2021-09-20T20:37:09.391Z',
        
                };
                const component = renderer.create(<DomainInfoCard data={input} />);
                let tree = component.toJSON();
                expect(tree).toMatchSnapshot();
                // On the first run of this test, Jest will generate a snapshot file automatically.
            });
      });

      describe ('renders correctly', () => {
        it('with missing nameservers', () => {
            let input = { 
                domainName: 'DomainName',
                status: 'status1 status2 status3 status4',
                expiresDate : '2021-09-20T20:37:09.391Z',
                createdDate : '2021-09-20T20:37:09.391Z',
    
            };
            const component = renderer.create(<DomainInfoCard data={input} />);
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
            // On the first run of this test, Jest will generate a snapshot file automatically.
        });
        it('with one server', () => {
            let input = { 
                domainName: 'DomainName',
                status: 'status1 status2 status3 status4',
                nameServers : { hostNames: ['server1' ]},
                expiresDate : '2021-09-20T20:37:09.391Z',
                createdDate : '2021-09-20T20:37:09.391Z',
    
            };
            const component = renderer.create(<DomainInfoCard data={input} />);
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
            // On the first run of this test, Jest will generate a snapshot file automatically.
        });
  })
      
});