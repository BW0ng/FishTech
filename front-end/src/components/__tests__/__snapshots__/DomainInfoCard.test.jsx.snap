// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainInfoCard renders correctly with full data output 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Domain Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <strong>
        Name: 
      </strong>
      DomainName
      <br />
      <br />
      <strong>
        Domain Status: 
      </strong>
      <br />
      <span>
         
        status1
        <br />
      </span>
      <span>
         
        status2
        <br />
      </span>
      <span>
         
        status3
        <br />
      </span>
      <span>
         
        status4
        <br />
      </span>
      <br />
      <br />
      <strong>
        Nameservers: 
      </strong>
      <br />
      <span>
         
        server1
        <br />
      </span>
      <span>
         
        server2
        <br />
      </span>
      <span>
         
        server3
        <br />
      </span>
      <span>
         
        server4
        <br />
      </span>
      <br />
      <br />
      <h3>
        Dates
      </h3>
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
      <br />
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
    </div>
  </div>
</div>
`;

exports[`DomainInfoCard renders correctly with missing nameservers 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Domain Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <strong>
        Name: 
      </strong>
      DomainName
      <br />
      <br />
      <strong>
        Domain Status: 
      </strong>
      <br />
      <span>
         
        status1
        <br />
      </span>
      <span>
         
        status2
        <br />
      </span>
      <span>
         
        status3
        <br />
      </span>
      <span>
         
        status4
        <br />
      </span>
      <br />
      <br />
      <strong>
        Nameservers: 
      </strong>
      <br />
      <br />
      <br />
      <h3>
        Dates
      </h3>
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
      <br />
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
    </div>
  </div>
</div>
`;

exports[`DomainInfoCard renders correctly with missing one status 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Domain Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <strong>
        Name: 
      </strong>
      DomainName
      <br />
      <br />
      <strong>
        Domain Status: 
      </strong>
      <br />
      <span>
         
        status1
        <br />
      </span>
      <br />
      <br />
      <strong>
        Nameservers: 
      </strong>
      <br />
      <span>
         
        server1
        <br />
      </span>
      <span>
         
        server2
        <br />
      </span>
      <span>
         
        server3
        <br />
      </span>
      <span>
         
        server4
        <br />
      </span>
      <br />
      <br />
      <h3>
        Dates
      </h3>
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
      <br />
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
    </div>
  </div>
</div>
`;

exports[`DomainInfoCard renders correctly with missing statuses 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Domain Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <strong>
        Name: 
      </strong>
      DomainName
      <br />
      <br />
      <strong>
        Domain Status: 
      </strong>
      <br />
      <br />
      <br />
      <strong>
        Nameservers: 
      </strong>
      <br />
      <span>
         
        server1
        <br />
      </span>
      <span>
         
        server2
        <br />
      </span>
      <span>
         
        server3
        <br />
      </span>
      <span>
         
        server4
        <br />
      </span>
      <br />
      <br />
      <h3>
        Dates
      </h3>
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
      <br />
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
    </div>
  </div>
</div>
`;

exports[`DomainInfoCard renders correctly with one server 1`] = `
<div
  className="accordion-item"
>
  <h2
    className="accordion-header"
  >
    <button
      aria-expanded={false}
      className="accordion-button collapsed"
      onClick={[Function]}
      type="button"
    >
      Domain Information
    </button>
  </h2>
  <div
    aria-expanded={null}
    className="accordion-collapse collapse"
  >
    <div
      className="accordion-body"
    >
      <strong>
        Name: 
      </strong>
      DomainName
      <br />
      <br />
      <strong>
        Domain Status: 
      </strong>
      <br />
      <span>
         
        status1
        <br />
      </span>
      <span>
         
        status2
        <br />
      </span>
      <span>
         
        status3
        <br />
      </span>
      <span>
         
        status4
        <br />
      </span>
      <br />
      <br />
      <strong>
        Nameservers: 
      </strong>
      <br />
      <span>
         
        server1
        <br />
      </span>
      <br />
      <br />
      <h3>
        Dates
      </h3>
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
      <br />
      <strong>
        Registry Expiration: 
      </strong>
      2021-09-20T20:37:09.391Z
      <br />
    </div>
  </div>
</div>
`;
